### PR TITLE
Issue/343

### PR DESCRIPTION
**Pull request by minborg**
This PR adds a large number of tests for `Serializable` objects. It also adds fixes to that these `Serializable` objects are handled correctly.

### CODE REVIEW

**Review by RobAustin — APPROVED** [review]

File: src/test/java/net/openhft/chronicle/wire/WireTests.java [hunk 1]
```diff
@@ -129,6 +129,7 @@ public void testDate() {
         assertEquals(new Date(1234567890000L), wire.getValueIn()
                 .object(Date.class));
 
+        /* Not sure why this would work
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/wire/WireTests.java:132]: should resolve this rather than leaving commented code

File: pom.xml [hunk 2]
```diff
@@ -98,8 +99,35 @@
...
-            <artifactId>junit</artifactId>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>net.openhft</groupId>
+            <artifactId>chronicle-test-framework</artifactId>
+            <version>1.22ea2</version>
```
Comment (JerryShea) [inline pom.xml]: is this version now in bom?

**Review by JerryShea — APPROVED** [review]
Looks great! Assume you will squash merge or rebase -i to clean commit history?

File: pom.xml (hunk 2 above)
Comment (minborg) [reply pom.xml]: Yes. I will fix

**Pull request merged**